### PR TITLE
Mc crypto ring signature no alloc improvements

### PR DESCRIPTION
**Pull request by garbageslam**
This implements two changes discussed in a meeting on Wednesday.
This helps to move us closer towards making `mc-crypto-ring-signature` not depends on the `alloc` crate, so that it can be used more easily on a Trezor without requiring the rust standard allocator.

### CODE REVIEW

File: account-keys/types/src/traits.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+use mc_crypto_keys::RistrettoPublic;
+
+/// An object which has represents a subaddress, and has RingCT-style
```
Comment (sugargoat) [inline account-keys/types/src/traits.rs:3]:
```suggestion
/// An object which represents a subaddress, and has RingCT-style
```

File: crypto/ring-signature/src/onetime_keys.rs [hunk 2]
```diff
@@ -98,7 +98,7 @@ fn hash_to_scalar(point: RistrettoPoint) -> Scalar {
 /// * `recipient` - The recipient subaddress `(C,D)`.
 pub fn create_tx_out_target_key(
     tx_private_key: &RistrettoPrivate,
-    recipient: &PublicAddress,
+    recipient: &impl RingCtAddress,
```
Comment (sugargoat) [inline crypto/ring-signature/src/onetime_keys.rs:101]: Does this become a breaking change for clients? 

I see it being used in full-service (tests only)

https://github.com/mobilecoinofficial/full-service/blob/f1e58e60e6915710bc45210643dc088dde379a84/full-service/src/test_utils.rs#L459

**Review by sugargoat — APPROVED** [review]
LGTM!

File: crypto/ring-signature/src/onetime_keys.rs (hunk 2 above)
Comment (garbageslam) [reply crypto/ring-signature/src/onetime_keys.rs:101]: I think it's okay because the current object they are passing implements this trait

**Review by nick-mobilecoin — APPROVED** [review]
LGTM

Still spinning up on github, I scratched my head for a bit wondering how the use statements for the `GeneratorCache` worked before.
glossed over the rename in the UI
image

**Pull request merged**